ros2doctor/ros2doctor/api/package.py

```diff
@@ -97,43 +97,51 @@ def compare_versions(result: Result, local_packages: dict, distro_packages: dict
     :param: boolean value determines which output to populate, msgs or report
     :return: list of warning messages
     """
-    missing_req = ''
-    missing_local = ''
+    missing_req = []
+    missing_local = []
     for name, local_ver_str in local_packages.items():
         if not local_ver_str:
-            missing_local += ' ' + name
+            missing_local += name
             local_ver_str = ''
         latest_ver_str = distro_packages.get(name, '')
         if not latest_ver_str:
-            missing_req += ' ' + name
-        local_ver = version.parse(local_ver_str).base_version
-        latest_ver = version.parse(latest_ver_str).base_version
+            missing_req += name
+
+        try:
```

Comment:
I like having this `try..except` block to deal with arbitrary version parsing errors.

However, if either `local_ver_str` or `latest_ver_str` is empty, there is really no reason to bother here at all; we know it is going to fail.  I think we should just skip the version parsing and comparison in that case.